schema.ts

```diff
@@ -1232,6 +1232,82 @@ export interface ObjectSchemaDefinition<K extends string, L extends string>
   // TODO(dweitzman): Only support options in the typing when the codaType is ValueHintType.SelectList.
 }
 
+/**
+ * Discriminent used to determine the permission type
+ * @hidden
+ */
+export enum PrincipalType {
+  User = 'user',
+  Group = 'group',
+  Domain = 'domain',
+  Public = 'public',
+}
+
+/**
+ * This type definition represents properties required for all permissions.
+ *
+ * TODO(sam): Unhide this
+ * @hidden
+ */
+interface BasePermission {
+  type: PrincipalType;
+  rowId: string | number;
+  // TODO(sam): Add options describing inheritance of permissions
+}
+
+/**
+ * This grants access to a specific user by their user ID.
+ *
+ * TODO(sam): Unhide this
+ * @hidden
+ */
+interface UserPermission extends BasePermission {
+  type: PrincipalType.User;
+  userId: string | number;
+}
+
+/**
+ * This grants access to a specific group by its group ID.
+ *
+ * TODO(sam): Unhide this
+ * @hidden
+ */
+interface GroupPermission extends BasePermission {
+  type: PrincipalType.Group;
+  groupId: string | number;
```

Comment:
depending on the result of the slack thread, we might want to bring back groupType

Reply:
Yeah

Reply:
Well, I mean there's a possibility

Reply:
yeah I am debating if we can make the assumption that the group id is unique enough. Obviously since the pack maker is in control of everything you _could_ always do something like `groupId: groupTypeOne-id` to represent this if we just had id

Reply:
oh, that's interesting. yeah then leave groupType out for now.